compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/original-reactive-scopes-fork/mutate-outer-scope-within-value-block.expect.md

```diff
@@ -68,26 +68,20 @@ import { CONST_TRUE, identity, shallowCopy } from "shared-runtime";
  * should be merged.
  */
 function useFoo(t0) {
-  const $ = _c(3);
+  const $ = _c(2);
   const { input } = t0;
   const arr = shallowCopy(input);
+
+  const cond = identity(false);
   let t1;
-  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
-    t1 = identity(false);
-    $[0] = t1;
-  } else {
-    t1 = $[0];
-  }
-  const cond = t1;
-  let t2;
-  if ($[1] !== arr) {
-    t2 = cond ? { val: CONST_TRUE } : mutate(arr);
-    $[1] = arr;
-    $[2] = t2;
+  if ($[0] !== arr) {
+    t1 = cond ? { val: CONST_TRUE } : mutate(arr);
```

Comment:
This output is invalid — note that `arr` gets mutated in the memo block, but declared outside of it — but that's because this is a bug that is fixed in the new HIR-based reactive scopes mode (which is on by default, this fixture opts out with `@enableReactiveScopesInHIR:false`). The version of this fixture using HIR-based reactive scopes doesn't change as a result of this PR, and both the `arr = shallowCopy(input)` and `identity(false)` are correctly inside the memo block for `arr`